payu/models/cesm_cmeps.py

```diff
@@ -163,8 +161,30 @@ def setup(self):
             )
         for realm in all_realms:
             ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
-            assert cpucount >= ntasks, "Insufficient cpus for the pelayout in nuopc.runconfig"
-        
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+            #rootpe is zero-based
+            assert cpucount >= (rootpe + ntasks*pestride), f"Insufficient cpus for the {realm} pelayout in nuopc.runconfig"
+
+        # check iolayout
+        for realm in self.realms:
+            #med and cpl names are both used in runconfig
+            if realm == "cpl" : realm="MED"
+            io_section = f"{realm.upper()}_modelio"
```

Comment:
@dougiesquire - Do you know, in nuopc.runconfig, are the `_modelio` names case-sensitive?

i.e. will it always be `ICE_modelio` or is `ice_modelio` valid ?

Reply:
I don't know sorry